eessi/testsuite/tests/apps/osu.py

```diff
@@ -0,0 +1,296 @@
+"""
+This module tests the binary 'osu' in available modules containing substring 'OSU-Micro-Benchmarks'.
+The basic application class is taken from the hpctestlib to which extra features are added.
+"""
+
+import os
+import reframe as rfm
+from reframe.core.meta import parameters
+import reframe.utility.sanity as sn
+
+from hpctestlib.microbenchmarks.mpi.osu import osu_benchmark
+
+from eessi.testsuite import hooks, utils
+from eessi.testsuite.constants import SCALES, TAGS, DEVICE_TYPES
+
+def my_filtering_function():
+    """
+    Filtering function for filtering scales for the pt2pt OSU test
+    """
+    scale_filtered = SCALES.copy()
+    for key in list(SCALES):
+        if(key == '1_core' or key == '4_cores' or
+           SCALES.get(key).get('num_nodes') > 2):
+            scale_filtered.pop(key)
+        elif('node_part' in SCALES.get(key)):
+            if(SCALES.get(key).get('node_part') > 1):
+                scale_filtered.pop(key)
+    return scale_filtered
+
+
+def my_filtering_function_coll():
+    """
+    Filtering function for filtering scales for collective the OSU test
+    """
+    scale_filtered = SCALES.copy()
+    for key in list(SCALES):
+        if(key == '1_core'):
+            scale_filtered.pop(key)
+    return scale_filtered
```

Comment:
same here:
```suggestion
    return [
        k for (k, v) in SCALES.items()
        if v['num_nodes'] * v.get('num_cpus_per_node', 1) > 1
    ]
```
which returns:
```
['2_cores',
 '4_cores',
 '1_cpn_2_nodes',
 '1_cpn_4_nodes',
 '2_nodes',
 '4_nodes',
 '8_nodes',
 '16_nodes']

```

Reply:
This one still misses node fractions. (half, 1/4th or 1/8th)